### PR TITLE
Remove viewfile and replace xdoc with xmosdoc

### PR DESCRIPTION
**Pull request by lucianomartin**
The docs have been updated just in order to generate the existing ones with xmosdoc. The documents must still be reworked.

The tests have been disabled, they must be reworked as part of  https://xmosjira.atlassian.net/issues/LSM-129.

Closes https://xmosjira.atlassian.net/browse/LSM-119
Part of https://xmosjira.atlassian.net/issues/LSM-120: introduce xmosdoc

### CODE REVIEW

File: doc/index.rst
Comment (xalbertoisorna) [inline doc/index.rst:1]: after new library policies, if possible avoid index.rst at all using new xmosdoc. 
see: https://github.com/xmos/lib_awe

File: examples/spi/README.rst [hunk 1]
```diff
@@ -33,9 +33,10 @@ Prerequisites
    aspects which are not specific to this application note are linked to in
    the references appendix.
 
- * For a description of XMOS related terms found in this document
+.. TODO: This link is not reachable. Remove or update it.
```
Comment (xalbertoisorna) [inline examples/spi/README.rst:36]: if these are comments, I believe it is better to remove them or leave a comment about what needs to be added.

File: lib_device_control/module_build_info [hunk 2]
```diff
@@ -1,7 +1,7 @@
 VERSION = 5.0.0
 
-DEPENDENT_MODULES = lib_xassert(>=4.0.0) \
-                    lib_logging(>=3.0.0)
+DEPENDENT_MODULES = lib_xassert(>=4.2.0) \
```
Comment (xalbertoisorna) [inline lib_device_control/module_build_info:3]: pin versions here

File: settings.yml
Comment (xalbertoisorna) [inline settings.yml:1]: nice, please add readme.st with orphan. check https://github.com/xmos/lib_awe

**Review by xalbertoisorna — CHANGES_REQUESTED** [review]
good PR, I would like to request few changes to adapt to latest library practices
https://xmosjira.atlassian.net/wiki/x/H4DE7Q

File: Jenkinsfile [hunk 3]
```diff
@@ -21,8 +21,9 @@ pipeline {
 
   environment {
     REPO = 'lib_device_control'
-    VIEW = getViewName(REPO)
-  }
+    XMOSDOC_VERSION = 'v5.5.1'
```
[the remark this answers is not in the document]
Comment (lucianomartin) [reply Jenkinsfile]: I guess, you meant 6.1.1

File: examples/spi/README.rst (hunk 1 above)
Comment (lucianomartin) [reply examples/spi/README.rst]: removed

File: lib_device_control/module_build_info (hunk 2 above)
Comment (lucianomartin) [reply lib_device_control/module_build_info:3]: This is not possible, see the error below during the library checks:

```
[2024-10-10T15:02:16.159Z]     def test_dependencies(dir):
[2024-10-10T15:02:16.159Z]         (res, msg) = do_deps_check(dir)
[2024-10-10T15:02:16.159Z] >       assert res, msg
[2024-10-10T15:02:16.159Z] E       AssertionError: Dependency lib_xassert(4.2.0) does not specify version
[2024-10-10T15:02:16.159Z] E       Dependency lib_logging(3.2.0) does not specify version
[2024-10-10T15:02:16.159Z] E       
[2024-10-10T15:02:16.159Z] E       module_build_info should specify dependencies with minimum versions e.g.
[2024-10-10T15:02:16.159Z] E       
[2024-10-10T15:02:16.159Z] E       DEPENDENT_MODULES = lib_logging(>=1.0.0) lib_ethernet(>=3.0.0)
```

File: examples/usb/doc/rst/usb.rst [hunk 4]
```diff
@@ -161,7 +161,8 @@ resource, setting LEDs or reading the buttons.
 
 Please see the source code listing app_listing_ for more details.
 
-|appendix|
+.. TODO: Check how the substitution below worked and update it.
```
Comment (mbanth) [inline examples/usb/doc/rst/usb.rst:164]: I have seen a few ToDo's.  They need to be resolved before merging or have a story created that includes them.

File: requirements.txt [hunk 5]
```diff
@@ -1,35 +1,2 @@
...
-#
-# If python code in this repository depends on python code under development
-# in another repository, then an entry for that other respository should
-# appear in this list instead of the released dependencies list.
-#
-# If this repository uses the setup functionality (e.g., script entry points)
-# of its own setup.py file, then this list must include an entry for that
-# setup.py file, e.g., '-e .' or '-e ./python' (without the quotes).
--e ./python
+# python_version 3.9
```
Comment (mbanth) [inline requirements.txt:1]: Does this repo use Python 3.9 consistently throughout?  Could it use a more recent version?

File: doc/substitutions.inc [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+.. _The XMOS XS3 Architecture: https://www.xmos.com/download/The-XMOS-XS3-Architecture.pdf
+.. |newpage| raw:: pdf
```
Comment (mbanth) [inline doc/substitutions.inc:3]: This looks like a multi-line substitution.  Have I interpreted it correctly and does it actually work?

**Review by mbanth — APPROVED** [review]
I have a few questions, and I have noted that some ToDo's remain unresolved.  However, the bulk of the changes look good.

File: settings.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+project: lib_device_control
+version: 5.0.0
+
+## example of overridable setting
+# watermarks is redefined in documentation and again for a specific product
+# eg different products might have different setting but we'd like to provide defaults
+watermarks: []
+
+products:
```
Comment (xalbertoisorna) [inline settings.yml:9]: I would not put a product, as it is a library, it does not need different document generations. use lib_name directly. 
check:
https://github.com/xmos/lib_awe/blob/develop/settings.yml

**Review by xalbertoisorna — CHANGES_REQUESTED** [review]
Looks goos, the only thing I would like to request a change on the settings.yml to folow library recomendations. https://xmosjira.atlassian.net/wiki/x/H4DE7Q

File: doc/substitutions.inc (hunk 6 above)
Comment (lucianomartin) [reply doc/substitutions.inc]: The substitution was not working, reading online we need some special packages such as rst2pdf to make it work. 
I removed all the "newpage" substitutions, as I couldn't make it work even without substitutions.

In [LSM-120]( https://xmosjira.atlassian.net/browse/LSM-120) we need to rewrite the documentation, so the page breaks will change too.

File: requirements.txt (hunk 5 above)
Comment (lucianomartin) [reply requirements.txt:1]: https://github.com/xmos/lib_device_control/pull/112/files will update the requirements file and use Python 3.10.

File: examples/usb/doc/rst/usb.rst (hunk 4 above)
Comment (lucianomartin) [reply examples/usb/doc/rst/usb.rst:162]: [LSM-120](https://xmosjira.atlassian.net/browse/LSM-120) covers rewriting the documentation.
The readme files should be updated there.

File: Jenkinsfile [hunk 8]
```diff
@@ -66,28 +87,41 @@ pipeline {
...
+        stage('Build documentation') {
+          agent {
+            label 'documentation'
+          }
+          stages {
+            stage('Docs') {
               steps {
-                dir("${REPO}/${REPO}") {
-                  runXdoc('doc')
+                createVenv("requirements.txt")
```
Comment (xalbertoisorna) [inline Jenkinsfile:105]: I prefer  https://github.com/xmosnotes/an02017/blob/073d9d1c05591f36eff9a7286b39bfe99975aa16/Jenkinsfile#L8
but also fine
`
def buildDocs(String zipFileName) {
    withVenv {
        sh 'pip install git+ssh://git@github.com/xmos/xmosdoc@v6.1.1'
        sh 'xmosdoc'
        zip zipFile: zipFileName, archive: true, dir: 'doc/_build'
    }
}


`

**Review by xalbertoisorna — APPROVED** [review]
looks good to me

File: Jenkinsfile (hunk 8 above)
Comment (lucianomartin) [reply Jenkinsfile:114]: I used a similar function after I cleaned it up a bit. The changes are in https://github.com/xmos/lib_device_control/pull/111/commits/ and https://github.com/xmos/lib_device_control/pull/111/commits/4b44aed7b7a20db44452529c37959aeb3acddc13.

**Pull request merged**